source/kraken/realtime.cpp

```diff
@@ -320,23 +320,61 @@ static bool is_handleable(const transit_realtime::TripUpdate& trip_update,
     // departure is brought forward), we check the size of stop_time_update because we can't find a proper
     // enum in gtfs-rt proto to express this idea
     if (is_circulating_trip(trip_update)) {
-        // check if company id exists
-        if (is_added_trip(trip_update) && trip_update.trip().HasExtension(kirin::company_id)
-            && pt_data.companies_map.find(trip_update.trip().GetExtension(kirin::company_id))
-                   == pt_data.companies_map.end()) {
-            LOG4CPLUS_DEBUG(log, "Trip company id " << trip_update.trip().GetExtension(kirin::company_id)
-                                                    << " doesn't exist: ignoring trip update id "
-                                                    << trip_update.trip().trip_id());
-            return false;
-        }
-        // check if physical mode id exists
-        if (is_added_trip(trip_update) && trip_update.vehicle().HasExtension(kirin::physical_mode_id)
-            && pt_data.physical_modes_map.find(trip_update.vehicle().GetExtension(kirin::physical_mode_id))
-                   == pt_data.physical_modes_map.end()) {
-            LOG4CPLUS_DEBUG(log, "Trip physical mode id " << trip_update.vehicle().GetExtension(kirin::physical_mode_id)
-                                                          << " doesn't exist: ignoring trip update id "
-                                                          << trip_update.trip().trip_id());
-            return false;
+        if (is_added_trip(trip_update)) {
+            // check if company id exists
+            if (trip_update.trip().HasExtension(kirin::company_id)
```

Comment:
minor: it'll be much more readable you can put all these codes in functions... like `check_company_id()` `check_physical_mode_id()`

besides, these blocs of codes look quite similar...